ledger/src/shred.rs

```diff
@@ -802,6 +835,7 @@ pub mod layout {
         }
     }
 
+    #[cfg_attr(feature = "dev-context-only-utils", qualifiers(pub))]
```

Comment:
The point of avoiding `pub` is that we can change this function api or behavior without the external code breaking.
If some tests in another crate is going to break then it is already a lost cause.
So I don't see much value in using `dev-context` here;  you can just make the function `pub`.